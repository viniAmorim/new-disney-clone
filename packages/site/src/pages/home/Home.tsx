import React, { BaseHTMLAttributes } from 'react';
import { UseFormReturn } from 'react-hook-form';
import { Col, Container, Row } from 'react-grid-system';
import { FaPlus } from 'react-icons/fa';

import {
  FormControl,
  Input,
  Heading,
  Button,
  Link,
  Paragraph,
} from '@site/uikit';
import NLink from 'next/link';

import * as Styled from './styles';
import { LoginFormValues } from './HomeContainer';

interface HomeProps extends BaseHTMLAttributes<HTMLDivElement> {
  onLogin: (values: LoginFormValues) => void;
  form: UseFormReturn;
  isLoading?: boolean;
}

function Home({ onLogin, form, isLoading = false, ...props }: HomeProps) {
  return (
    <Styled.Background {...props}>
      <Container>
        <Row>
          <Col sm={6} md={6}>
            <Styled.Logo src="/images/logo.png" alt="Correta.app" />
            <Styled.StyledHeading size="headline">Entrar</Styled.StyledHeading>
            <form onSubmit={form.handleSubmit(onLogin)}>
              <FormControl
                label="E-mail:"
                error={form.formState.errors?.email?.message}
              >
                <Input
                  name="email"
                  placeholder="seu@email.com"
                  {...form.register('email', {
                    required: true,
                    pattern: {
                      value: /\S+@\S+\.\S+/,
                      message: 'E-mail inválido.',
                    },
                    minLength: {
                      value: 5,
                      message: 'E-mail é obrigatório',
                    },
                  })}
                />
              </FormControl>
              <FormControl
                label="Senha:"
                error={form.formState.errors?.password?.message}
              >
                <Input
                  data-testid="password-input"
                  name="password"
                  placeholder="senha"
                  type="password"
                  {...form.register('password', {
                    required: true,
                    minLength: {
                      value: 5,
                      message: 'Senha é obrigatória',
                    },
                  })}
                />
              </FormControl>
              <Styled.ForgotPasswordContainer>
                <Button
                  data-testid="login-button"
                  type="submit"
                  isLoading={isLoading}
                >
                  Entrar
                </Button>
              </Styled.ForgotPasswordContainer>
            </form>
          </Col>
          <Styled.ContentCol sm={6} md={6}>
            <Styled.ContentContainer>
              <img src="/images/home.png" alt="Correta.app" />
              <span>
                <FaPlus />
                <Heading size="display">facilidade</Heading>
              </span>
              <span>
                <FaPlus />
                <Heading size="display">resultados</Heading>
              </span>
              <Paragraph size="headline">
                Tenha acesso e controle de seus cliente, melhorando seu processo
                de vendas e gerando mais resultados.
              </Paragraph>
            </Styled.ContentContainer>
          </Styled.ContentCol>
        </Row>
      </Container>
    </Styled.Background>
  );
}

export default Home;
