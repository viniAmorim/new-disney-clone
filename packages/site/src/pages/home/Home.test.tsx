import React from 'react';

import userEvent from '@testing-library/user-event';

import { fireEvent, render, waitFor } from '~/test/utils';

import Home from './Home';

const onLogin = jest.fn();

describe('Home', () => {
  it('should render correctly', async () => {
    const { queryByTestId } = render(
      <Home data-testid="home" onLogin={onLogin} />,
    );

    const home = queryByTestId('home');

    expect(home).toBeInTheDocument();
    expect(home).toMatchSnapshot();
  });

  it('should show error when email is empty', async () => {
    const { getByTestId, getByText } = render(<Home onLogin={onLogin} />);

    const button = getByTestId('login-button');
    fireEvent.click(button);

    await waitFor(() => expect(getByText('E-mail é obrigatório')).toBeTruthy());
  });

  it('should show error when email is empty', async () => {
    const { getByRole, getByTestId, getByText } = render(
      <Home onLogin={onLogin} />,
    );

    const emailInput = getByRole('textbox');
    userEvent.type(emailInput, 'invalid-email.com');

    const button = getByTestId('login-button');
    fireEvent.click(button);

    await waitFor(() => expect(getByText('E-mail inválido')).toBeTruthy());
  });

  it('should show error when password is empty', async () => {
    const { getByRole, getByTestId, getByText } = render(
      <Home onLogin={onLogin} />,
    );

    const emailInput = getByRole('textbox');
    userEvent.type(emailInput, 'email@email.com');

    const button = getByTestId('login-button');
    fireEvent.click(button);

    await waitFor(() => expect(getByText('Senha é obrigatória')).toBeTruthy());
  });

  it('should get login data when clicked in login button', async () => {
    const { getByRole, getByTestId } = render(<Home onLogin={onLogin} />);

    const emailInput = getByRole('textbox');
    userEvent.type(emailInput, 'email@email.com');

    const passwordInput = getByTestId('password-input');
    userEvent.type(passwordInput, 'password');

    const button = getByTestId('login-button');
    fireEvent.click(button);

    await waitFor(() => {
      expect(onLogin.mock.calls[0][0]).toEqual({
        email: 'email@email.com',
        password: 'password',
      });
    });
  });
});
