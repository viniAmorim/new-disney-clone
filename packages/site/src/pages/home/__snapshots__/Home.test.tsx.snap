// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div
  class="styles__Background-sc-ckx671-2 jIWJlP"
  data-testid="home"
>
  <div
    style="box-sizing: border-box; position: relative; margin-left: auto; margin-right: auto; padding-left: 15px; padding-right: 15px; max-width: 960px;"
  >
    <div
      style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start; flex-direction: row;"
    >
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;"
      >
        <img
          alt="Correta.app"
          class="styles__Logo-sc-ckx671-0 iHUYHr"
          src="/images/logo.png"
        />
        <h1
          class="sc-gKAaRy kbUCYm styles__StyledHeading-sc-ckx671-1 dCoeDK"
        >
          Entrar
        </h1>
        <form>
          <div
            class="sc-hKFxyN zFDVE"
          >
            <label
              class="sc-eCApnc CDjog"
            >
              E-mail:
            </label>
            <input
              class="sc-iCoGMd kmeLFP"
              name="email"
              placeholder="seu@email.com"
              value=""
            />
          </div>
          <div
            class="sc-hKFxyN zFDVE"
          >
            <label
              class="sc-eCApnc CDjog"
            >
              Senha:
            </label>
            <input
              class="sc-iCoGMd kmeLFP"
              data-testid="password-input"
              name="password"
              placeholder="senha"
              type="password"
              value=""
            />
          </div>
          <div
            class="styles__ForgotPasswordContainer-sc-ckx671-5 hkWuNd"
          >
            <a
              class="sc-pNWdM iFoZxv"
            >
              Esqueci minha senha
            </a>
            <button
              class="sc-gtsrHT kGzhap"
              data-testid="login-button"
              type="submit"
            >
              Entrar
            </button>
          </div>
          <div
            class="styles__RegisterContainer-sc-ckx671-6 fYjycG"
          >
            <p
              class="sc-iemWCZ eTebHM"
            >
              Ainda n√£o tem cadastro?
            </p>
            <button
              class="sc-gtsrHT gBmKmV"
            >
              Cadastre-se gratuitamente
            </button>
          </div>
        </form>
      </div>
      <div
        class="styles__ContentCol-sc-ckx671-3 eNHZgk"
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;"
      >
        <div
          class="styles__ContentContainer-sc-ckx671-4 iIMLuo"
        >
          <img
            alt="Correta.app"
            src="/images/home.png"
          />
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              />
            </svg>
            <h1
              class="sc-gKAaRy eeYKHx"
            >
              facilidade
            </h1>
          </span>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              />
            </svg>
            <h1
              class="sc-gKAaRy eeYKHx"
            >
              resultados
            </h1>
          </span>
          <p
            class="sc-iemWCZ bCVkDu"
          >
            Tenha acesso e controle de seus cliente, melhorando seu processo de vendas e gerando mais resultados.
          </p>
        </div>
      </div>
    </div>
    <span
      style="display: table; clear: both;"
    />
  </div>
</div>
`;
